### PR TITLE
Added notebook showing regression variants

### PR DESCRIPTION
**Pull request by sdraper-CS**
See issue 2872 in main Certifai repo

### CODE REVIEW

File: notebooks/regression/Regression-Variants.ipynb [hunk 1]
```diff
@@ -0,0 +1,3680 @@
...
+    "Licensed under CognitiveScale Example Code [License](https://github.com/CognitiveScale/cortex-certifai-examples/blob/master/LICENSE.md)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Alternate formulations for regression \n",
+    "\n",
+    "In this notebook we'll show how to modify the formulation of what constitutes a 'significant change' for the purposes of deciding how much change is required for an outcome to be considered cifferent (counterfactual) to the original data point's prediction.\n",
```
Comment (cdraper-CS) [inline notebooks/regression/Regression-Variants.ipynb:17]: cifferent => different

File: notebooks/regression/Regression-Variants.ipynb [hunk 2]
```diff
@@ -0,0 +1,3680 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the favorable direction to be increasing, and we'll consider\n",
+    "# predictions above the mean-prediciton for\n",
```
Comment (cdraper-CS) [inline notebooks/regression/Regression-Variants.ipynb:211]: prediciton

File: notebooks/regression/Regression-Variants.ipynb [hunk 3]
```diff
@@ -0,0 +1,3680 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Set up percentile threshold formulation\n",
+    "\n",
+    "Similarly we can define a fixed threshold in terms of percentiles of the evaluation dataset's predicted values.\n",
+    "In this case let's define 'favorable' as the top 20% of th distribution."
```
Comment (cdraper-CS) [inline notebooks/regression/Regression-Variants.ipynb:1948]: th => the

**Comment by cdraper-CS** [comment]
Does it seem surprising that for the 80th percentile favorable, the LinL1 model has the favorable ones over 600, whereas the SVM model has the favorable ones over 560? I think the counter-intuitive piece is... 80% favorable may be different for different models, right? Would it be easy to calculate/show the 80% level for the two models and point this out?

**Comment by sdraper-CS** [comment]
@cdraper-CS Addressed your comments

File: notebooks/regression/Regression-Variants.ipynb [hunk 4]
```diff
@@ -3654,6 +3654,25 @@
...
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Notes\n",
+    "\n",
+    "It is apparent from the last formulation that using a percentile of the outcome distribution is *not* the\n",
+    "same as specifying an a-priori fixed value as the favorability threshold.  This is apparent by comparing the results of the two models which show the SVM model had an 80% percencile at about \\\\$560, whereas the logistic model had one at a little over \\\\$600.  It is also noteable that the logistic model predicts some negative values, which are clearly not realistic.  Both of these are indications that the logistic model (which will have a linear decision surface) is insufficiently complex to capture the data behaviour at the extremes.  This is despite its R-squared actually being better than that of the SVM.\n",
```
Comment (cdraper-CS) [inline notebooks/regression/Regression-Variants.ipynb:3665]: percencile

Comment (sdraper-CS) [reply notebooks/regression/Regression-Variants.ipynb]: Fixed

**Review by cdraper-CS — APPROVED** [review]

**Pull request merged**